Clean up go test resources to enable repeatable tests

This is existing behaviour with the tests. Use `t.Cleanup`

```
➜  goose git:(master) go test
2021/10/28 17:52:06 Created new file: /var/folders/86/m_5387nn3wl47ljp02vg923c0000gn/T/test_create_osfs2668887497/20211028175206_test.sql
2021/10/28 17:52:06 RENAMED 20211028175206_test.sql => 00001_test.sql
--- FAIL: TestEmbeddedMigrations (0.00s)
    --- FAIL: TestEmbeddedMigrations/Migration_cycle (0.00s)
        goose_test.go:128: Failed to run 'up' migrations: error: found 2 missing migrations:
            	version 1: 00001_create_users_table.sql
            	version 2: 00002_rename_root.sql
        goose_test.go:137: Expected version 3 after 'up', got 0
FAIL
exit status 1
FAIL	github.com/pressly/goose/v3	11.180s
```

--- a/.gitignore
+++ b/.gitignore
@@ -5,4 +5,3 @@
 
 # Files output by tests
 /bin
-/*.db

--- a/goose_test.go
+++ b/goose_test.go
@@ -26,7 +26,14 @@ func TestDefaultBinary(t *testing.T) {
 		"./bin/goose -dir=examples/sql-migrations sqlite3 sql.db status",
 		"./bin/goose --version",
 	}
-	defer os.Remove("./bin/goose") // clean up
+	t.Cleanup(func() {
+		if err := os.Remove("./bin/goose"); err != nil {
+			t.Logf("failed to remove %s resources: %v", t.Name(), err)
+		}
+		if err := os.Remove("./sql.db"); err != nil {
+			t.Logf("failed to remove %s resources: %v", t.Name(), err)
+		}
+	})
 
 	for _, cmd := range commands {
 		args := strings.Split(cmd, " ")
@@ -52,9 +59,14 @@ func TestLiteBinary(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-
-	defer os.RemoveAll(dir)             // clean up
-	defer os.Remove("./bin/lite-goose") // clean up
+	t.Cleanup(func() {
+		if err := os.RemoveAll(dir); err != nil {
+			t.Logf("failed to remove %s resources: %v", t.Name(), err)
+		}
+		if err := os.Remove("./bin/lite-goose"); err != nil {
+			t.Logf("failed to remove %s resources: %v", t.Name(), err)
+		}
+	})
 
 	// this has to be done outside of the loop
 	// since go only supports space separated tags list.
@@ -90,6 +102,11 @@ func TestCustomBinary(t *testing.T) {
 		"./bin/custom-goose -dir=examples/go-migrations sqlite3 go.db down",
 		"./bin/custom-goose -dir=examples/go-migrations sqlite3 go.db status",
 	}
+	t.Cleanup(func() {
+		if err := os.Remove("./go.db"); err != nil {
+			t.Logf("failed to remove %s resouces: %v", t.Name(), err)
+		}
+	})
 
 	for _, cmd := range commands {
 		args := strings.Split(cmd, " ")
@@ -105,11 +122,15 @@ var migrations embed.FS
 
 func TestEmbeddedMigrations(t *testing.T) {
 	// not using t.Parallel here to avoid races
-
 	db, err := sql.Open("sqlite3", "sql_embed.db")
 	if err != nil {
 		t.Fatalf("Database open failed: %s", err)
 	}
+	t.Cleanup(func() {
+		if err := os.Remove("./sql_embed.db"); err != nil {
+			t.Logf("failed to remove %s resouces: %v", t.Name(), err)
+		}
+	})
 
 	db.SetMaxOpenConns(1)
 

